### PR TITLE
Updates tests to use assertThat instead of assertEquals

### PR DESCRIPTION
**Pull request by pavanbaloju**
# pull request for (#2147 )

In this Pr, all the below tests have been updated to use assertj
1.  ComplexNumberTest.java
2. EtlTest.java
3. FlattenerTest.java
4. FoodChainTest.java
5. GigasecondTest.java
6. GoCountingTest.java
7. GrainsTest.java
8. GrepToolTest.java
9. IsbnVerifierTest.java
10. LargestSeriesProductCalculatorTest.java
11. MatrixTest.java
12. MeetupTest.java
13. MicroBlogTest.java
14. MinesweeperBoardTest.java
15. NucleotideCounterTest.java
16. OctalTest.java
17. PascalsTriangleGeneratorTest.java
18. NaturalNumberTest.java
19. PigLatinTranslatorTest.java
20. PokerTest.java
21. PovTest.java
22. PrimeFactorsCalculatorTest.java
23. PythagoreanTripletTest.java

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/practice/octal/src/test/java/OctalTest.java
Comment (sanderploegsma) [inline exercises/practice/octal/src/test/java/OctalTest.java:1]: This exercise has been deprecated, so please revert these changes.

Note: for future reference you can see the status of each exercise in `config.json` at the root of this repo. This command will give you all the slugs for a quick check:

```sh
cat config.json | jq -r '.exercises.practice[] | select(.status == "deprecated") | .slug'
```

**Review by sanderploegsma — CHANGES_REQUESTED** [review]

File: exercises/practice/octal/src/test/java/OctalTest.java
Comment (pavanbaloju) [reply exercises/practice/octal/src/test/java/OctalTest.java:1]: Ahh, I was checking the config.json file but it seems I missed this one. Will revert it!

Comment (pavanbaloju) [reply exercises/practice/octal/src/test/java/OctalTest.java:1]: Done @sanderploegsma !

**Review by sanderploegsma — APPROVED** [review]
LGTM! Thanks for your contribution 🙏

**Comment by pavanbaloju** [comment]
> LGTM! Thanks for your contribution 🙏

Thank you!

**Pull request merged**